prime-router/docs/proposals/0020-univeral-pipeline-topics.md

```diff
@@ -0,0 +1,63 @@
+fields to be provided, and will likely be relevant to different sets of senders and receivers.
+
+## Goals
+
+Since ELR and ETOR have different purposes and their data is held to different expectations, we want organization
+settings to prevent the two data types (as well as future other data types) from getting mixed up between
+organizations (senders/receivers) that are intended for the other data type. The solution for this should be simple,
+foolproof, and independent as laid out here:
+
+- Simple: The Requirement is best described with an example: It is likely that ETOR onboarding and operations will be
+  managed by Technical Assistants (TA) hired directly by CDC, not by people on the ReportStream team. Those
+  non-ReportStream teams need a very simple foolproof way to stovepipe their data, to avoid confusion and errors.
+- Foolproof: The implementation should default to not send -- it can't depend on a human remembering to put in an extra
+  clause in one of the filters. (Although that solution may be fine for a UP MVP).
+- Independent of the data itself: You can easily conceive of situations where the data itself might not contain enough
+  info to determine if it belongs within a particular topic. Or, situations where the customers repeatedly change
+  datatypes as they set up their system. Therefore, the information that tells us where certain data goes should be tied
+  to senders and receivers, and may not be solely determinable from the data itself. Topics should be an abstraction.
+
+## Proposal
+
+Currently, the Universal Pipeline (UP) is used for both use cases, and organization settings specify to use the UP by
+setting the `topic` field to `full-elr`. This proposal suggests retaining the `full-elr` topic, but also adding a `etor`
+topic. Those topics will both signify usage of the UP, however senders for one topic will only be able to send data to
+receivers of the same topic. Additionally, each topic may have a different set of default receivers; for instance, the
+current default quality filter for the `full-elr` topic expects a patient to have contact information, but for newborn
+screening ETOR requests, that information won't be available. So a separate default quality filter for the `etor` topic
+could set a reasonable starting set of quality expectations that are reasonable for ETOR requests.
+
+### Implications on internal types
+
+Currently, senders in the code are represented by a few different types. The overall parent class is called `Sender`.
+`Sender` is an abstract class which is extended by `FullELRSender` (used for UP) and `TopicSender` (used for the legacy
+pipeline). `TopicSender` has subtypes `CovidSender` and `MonkeypoxSender`.
+
+Under this proposal, `FullELRSender` will be renamed to `UniversalPipelineSender`, and `TopicSender` will be renamed
+to `LegacyPipelineSender` (still with subclasses `CovidSender` and `MonkeypoxSender`). Topic will still be used to
```

Comment:
The refactoring of `TopicSender` may be better happening after all other parts of this proposal since there's no overlap with that name.

Reply:
Sure, a lot of that refactoring would be for clarity/readability only, so I think that would come after the introduction of an `etor` topic and any default filters that would exist for that topic.